WordPress/src/androidTest/java/org/wordpress/android/e2e/pages/ReaderViewPage.java

```diff
@@ -0,0 +1,106 @@
+package org.wordpress.android.e2e.pages;
+
+import android.view.KeyEvent;
+
+import androidx.test.platform.app.InstrumentationRegistry;
+import androidx.test.uiautomator.UiDevice;
+import androidx.test.uiautomator.UiObject;
+import androidx.test.uiautomator.UiSelector;
+
+import static junit.framework.TestCase.assertFalse;
+import static junit.framework.TestCase.assertTrue;
+import static org.wordpress.android.support.WPSupportUtils.DEFAULT_TIMEOUT;
+import static org.wordpress.android.support.WPSupportUtils.isTextDisplayed;
+import static org.wordpress.android.support.WPSupportUtils.swipeToLeft;
+import static org.wordpress.android.support.WPSupportUtils.swipeToRight;
+
+public class ReaderViewPage {
+    UiDevice mDevice = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation());
+
+    String mLikerContainerId = "org.wordpress.android.prealpha:id/liker_faces_container";
+    String mRelatedPostsId = "org.wordpress.android.prealpha:id/container_related_posts";
+
+    UiObject mLikerContainer = mDevice.findObject(new UiSelector().resourceId(mLikerContainerId));
+    UiObject mRelatedPostsContainer = mDevice.findObject(new UiSelector().resourceId(mRelatedPostsId));
+    UiObject mSwipeForMore = mDevice.findObject(new UiSelector().textContains("Swipe for more"));
+
+    public ReaderViewPage waitUntilLoaded() {
+        mRelatedPostsContainer.waitForExists(DEFAULT_TIMEOUT);
+
+        return this;
+    }
+
+    public ReaderViewPage like() {
+        tapLikeButton();
+        mLikerContainer.waitForExists(DEFAULT_TIMEOUT);
+
+        return this;
+    }
+
+    public ReaderViewPage unlike() {
+        tapLikeButton();
+        mLikerContainer.waitUntilGone(DEFAULT_TIMEOUT);
+
+        return this;
+    }
+
+    private void tapLikeButton() {
+        mSwipeForMore.waitUntilGone(DEFAULT_TIMEOUT);
+        // Even though it was working locally in simulator, tapping the footer buttons,
+        // like 'mLikeButton.click()', was not working in CI.
+        // The current workaround is to use arrows navigation.
+        mDevice.pressKeyCode(KeyEvent.KEYCODE_DPAD_DOWN);
+        mDevice.pressKeyCode(KeyEvent.KEYCODE_DPAD_RIGHT);
+        mDevice.pressKeyCode(KeyEvent.KEYCODE_DPAD_RIGHT);
+        mDevice.pressKeyCode(KeyEvent.KEYCODE_DPAD_RIGHT);
+        mDevice.pressKeyCode(KeyEvent.KEYCODE_DPAD_CENTER);
+        resetFocus();
+    }
+
+    private void resetFocus() {
+        mDevice.pressKeyCode(KeyEvent.KEYCODE_DPAD_LEFT);
+        mDevice.pressKeyCode(KeyEvent.KEYCODE_DPAD_LEFT);
+        mDevice.pressKeyCode(KeyEvent.KEYCODE_DPAD_LEFT);
+    }
+
+    public ReaderPage goBackToReader() {
```

Comment:
Should this be `ReaderViewPage` ?
```suggestion
    public ReaderViewPage goBackToReader() {
```

Reply:
Oh, `ReaderPage` is actually correct. It's were you land when viewing a post and clicking the back arrow at the top left corner.

![image](https://user-images.githubusercontent.com/42008628/149775635-f3bb7e0d-b7c5-4eec-8686-1406da20e86f.png)